common/component/azure/servicebus/client.go

```diff
@@ -38,14 +38,16 @@ type Client struct {
 	metadata    *Metadata
 	lock        *sync.RWMutex
 	senders     map[string]*servicebus.Sender
+	topicCache  map[string]bool
 }
 
 // NewClient creates a new Client object.
 func NewClient(metadata *Metadata, rawMetadata map[string]string) (*Client, error) {
 	client := &Client{
-		metadata: metadata,
-		lock:     &sync.RWMutex{},
-		senders:  make(map[string]*servicebus.Sender),
+		metadata:   metadata,
```

Comment:
The original issue was about GCP- why do we want to do it for Azure too?

Reply:
Its actually the function ```ensureTopic``` that has the potential to bottleneck right ? so if any service calling it can potentially endup bottle necking . Correct my thought process please